source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -2201,3 +2203,11 @@ pub(crate) fn closure_to_vir<'tcx>(
         panic!("closure_to_vir expects ExprKind::Closure");
     }
 }
+
+pub(crate) fn field_name_to_vir_ident(name: &str) -> vir::ast::Ident {
+    if name.bytes().nth(0).unwrap().is_ascii_digit() {
```

Comment:
Do we have to rely on the first character being a digit for this? Could rustc already have facilities for understanding this positional field assignment style?

Reply:
I'm not aware of one.

If we don't like it, it might be easier to just handle all fields - numeric and non-numeric - in the same way.

Reply:
right now we convert "0" -> "_0" but this is probably unnecessary. Nothing in our AIR encoding requires fields to start with a non-digit character.

Reply:
> right now we convert "0" -> "_0" but this is probably unnecessary. Nothing in our AIR encoding requires fields to start with a non-digit character.

This might be because SMTLIB atoms (e.g. field accessors function names) need to start with a [_A-Za-z] ? But it may be they are always prefixed by the name of the datatype anyway.

Reply:
I'm actually okay with this transformation as (if I understand correctly) it maps names to names and it does not have to match other code.

Reply:
Maybe a comment would be nice.

Reply:
It seems brittle to have the condition here that's almost (but not exactly) supposed to match https://github.com/verus-lang/verus/blob/main/source/rust_verify/src/rust_to_vir_adts.rs#L76-L82

Considering that we don't have the distinction from `rustc_middle` there, and here we can get the corresponding names as the initializer names, maybe we should indeed do this:

> it might be easier to just handle all fields - numeric and non-numeric - in the same way